addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCAnalogHandler.java

```diff
@@ -0,0 +1,285 @@
+import org.slf4j.LoggerFactory;
+
+import Moka7.S7;
+import Moka7.S7Client;
+
+/**
+ * The {@link PLCAnalogHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Falkenstern - Initial contribution
+ */
+@NonNullByDefault
+public class PLCAnalogHandler extends PLCCommonHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_ANALOG);
+
+    private final Logger logger = LoggerFactory.getLogger(PLCAnalogHandler.class);
+    private AtomicReference<PLCAnalogConfiguration> config = new AtomicReference<>();
+
+    private static final Map<String, @Nullable Integer> LOGO_BLOCKS_0BA7;
+    static {
+        Map<String, @Nullable Integer> buffer = new HashMap<>();
+        buffer.put("AI", 8); // 8 analog inputs
+        buffer.put("AQ", 2); // 2 analog outputs
+        buffer.put("AM", 16); // 16 analog markers
+        LOGO_BLOCKS_0BA7 = Collections.unmodifiableMap(buffer);
+    }
+
+    private static final Map<String, @Nullable Integer> LOGO_BLOCKS_0BA8;
+    static {
+        Map<String, @Nullable Integer> buffer = new HashMap<>();
+        buffer.put("AI", 8); // 8 analog inputs
+        buffer.put("AQ", 8); // 8 analog outputs
+        buffer.put("AM", 64); // 64 analog markers
+        buffer.put("NAI", 32); // 32 network analog inputs
+        buffer.put("NAQ", 16); // 16 network analog outputs
+        LOGO_BLOCKS_0BA8 = Collections.unmodifiableMap(buffer);
+    }
+
+    private static final Map<String, Map<String, @Nullable Integer>> LOGO_BLOCK_NUMBER;
+    static {
+        Map<String, Map<String, @Nullable Integer>> buffer = new HashMap<>();
+        buffer.put(LOGO_0BA7, LOGO_BLOCKS_0BA7);
+        buffer.put(LOGO_0BA8, LOGO_BLOCKS_0BA8);
+        LOGO_BLOCK_NUMBER = Collections.unmodifiableMap(buffer);
+    }
+
+    /**
+     * Constructor.
+     */
+    public PLCAnalogHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!isThingOnline()) {
+            return;
+        }
+
+        Channel channel = getThing().getChannel(channelUID.getId());
+        Objects.requireNonNull(channel, "PLCAnalogHandler: Invalid channel found");
+
+        PLCLogoClient client = getLogoClient();
+        String name = getBlockFromChannel(channel);
+
+        int address = getAddress(name);
+        if ((address != INVALID) && (client != null)) {
+            if (command instanceof RefreshType) {
+                int base = getBase(name);
+                byte[] buffer = new byte[getBufferLength()];
+                int result = client.readDBArea(1, base, buffer.length, S7Client.S7WLByte, buffer);
+                if (result == 0) {
+                    updateChannel(channel, S7.GetShortAt(buffer, address - base));
+                } else {
+                    logger.warn("Can not read data from LOGO!: {}.", S7Client.ErrorText(result));
```

Comment:
You do a lot of WARN logging throughout the code here. Note that normally you should only use DEBUG level logging. A WARN means that you have encountered a software bug that the user should report as an issue back to you. All "expected" situations, such as network issues that might happen at the user side should only result in the Thing to go OFFLINE with an according message.